src/internal/api/cluster/state.go

```diff
@@ -0,0 +1,39 @@
+package cluster
+
+import (
+	"net/http"
+
+	"github.com/defenseunicorns/zarf/src/internal/api/common"
+	"github.com/defenseunicorns/zarf/src/internal/k8s"
+	"github.com/defenseunicorns/zarf/src/internal/message"
+	"github.com/defenseunicorns/zarf/src/types"
+)
+
+// Read the zarf state secret from the cluster, if it exists.
+func ReadState(w http.ResponseWriter, r *http.Request) {
+	message.Debug("state.Read()")
+
+	data, err := k8s.LoadZarfState()
+	if err != nil {
+		message.ErrorWebf(err, w, "unable to load zarf state")
+	}
+
+	if data.Distro == "" {
+		common.WriteEmpty(w)
+	} else {
+		common.WriteJSONResponse(w, data, http.StatusOK)
+	}
+}
+
+// Update the zarf state secret in the cluster.
+func UpdateState(w http.ResponseWriter, r *http.Request) {
+	message.Debug("state.Update()")
+
+	var data types.ZarfState
+
+	if err := k8s.SaveZarfState(data); err != nil {
+		common.WriteJSONResponse(w, nil, http.StatusInternalServerError)
```

Comment:
This should use `message.ErrorfWebf` for conformity sake

Reply:
Even copilot knew this...

<img width="637" alt="Screen Shot 2022-09-19 at 12 08 32 PM" src="https://user-images.githubusercontent.com/882485/191073998-ad584b2f-bb71-4eac-9e5e-bba4971b9bbc.png">

Reply:
I, for one, welcome our future AI overlords.